CRM/Upgrade/Incremental/php/FiveFortySeven.php

```diff
@@ -30,11 +30,35 @@ class CRM_Upgrade_Incremental_php_FiveFortySeven extends CRM_Upgrade_Incremental
   public function upgrade_5_47_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
     $this->addTask('Migrate CiviGrant component to an extension', 'migrateCiviGrant');
+    $this->addTask('Add created_date to civicrm_relationship', 'addColumn', 'civicrm_relationship', 'created_date',
+      "timestamp NOT NULL  DEFAULT CURRENT_TIMESTAMP COMMENT 'Relationship created date'"
+    );
+    $this->addTask('Add modified_date column to civicrm_relationship', 'addColumn',
+      'civicrm_relationship', 'modified_date',
+      "timestamp NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP NOT NULL COMMENT 'Relationship last modified.'"
+    );
+    $this->addTask('Set initial value for relationship created_date and modified_date to start_date', 'updateRelationshipDates');
   }
 
   /**
    * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function updateRelationshipDates(CRM_Queue_TaskContext $ctx): bool {
+    CRM_Core_DAO::executeQuery('
+      UPDATE civicrm_relationship SET created_date = start_date, modified_date = start_date WHERE start_date IS NOT NULL
```

Comment:
I don't have a specific example where start_date is less than 1970-01-01 GMT but I've seen it elsewhere and know how to make it. This should then crash because timestamp can't be less than that, whereas start_date is a date which can be.

It's escaping me at the moment but something like `UPDATE ... SET created_date = MAX('1970-01-01', start_date)...` would work except there is no such SQL function.

Reply:
Good spotting - since the update is just updating to have a more useful starting value than 'right now' I added to the WHERE clause `      WHERE start_date IS NOT NULL AND start_date > "1970-01-01"`

Reply:
it's arguabe that I should update it at all - but I think it's fine with the update

Reply:
Yea I guess `start_date` often does correspond to when the relationship was created... but sometimes it doesn't. 
But for sites with advanced logging enabled, both created and updated dates should be available in the log tables somewhere. Wouldn't that be more accurate?

Reply:
@colemanw yes they are there - but it is a bit of a PITA to grab them - do you think it is that important to get the starting value right - it will be right going forwards & that will date quickly enough

Reply:
It's not that important imo, no

Reply:
OK _ I think this is mergeable - it adds the columns - but it doesn't expose them anywhere....